packages/editor/src/components/browser/TypeBrowser.tsx

```diff
@@ -175,12 +174,28 @@ const TypeBrowser = ({ onChange, onDoubleClick, initSearchFilter, location }: Ty
   });
 
   useEffect(() => {
-    const addListGeneric = (value: TypeBrowserObject, typeAsList: boolean) => {
-      if (typeAsList) {
-        return 'java.util.List<' + value.fullQualifiedName + '>';
+    const addListGeneric = (value: TypeBrowserObject, typeAsList: boolean, isIvyType: boolean, inCodeBlock: boolean) => {
+      let prefix = '';
+      let suffix = value.fullQualifiedName;
+      if (isIvyType) {
+        if (inCodeBlock) {
+          prefix = typeAsList ? 'List<' : '';
+          suffix = typeAsList ? value.simpleName + '>' : value.simpleName;
+        } else {
+          prefix = typeAsList ? 'java.util.List<' : '';
+          suffix = typeAsList ? value.simpleName + '>' : value.simpleName;
+        }
       } else {
-        return value.fullQualifiedName;
+        if (inCodeBlock) {
+          prefix = typeAsList ? 'List<' : '';
+          suffix = typeAsList ? value.fullQualifiedName + '>' : value.fullQualifiedName;
+        } else {
+          prefix = typeAsList ? 'java.util.List<' : '';
+          suffix = typeAsList ? value.fullQualifiedName + '>' : value.fullQualifiedName;
+        }
       }
+
+      return `${prefix}${suffix}`;
     };

```

Comment:
If you name this function 'addListGeneric' it should do exactly this, means, the typeAsList should not be a parameter, but a condition if this method is called or not. Then you can write this much simpler and much more readable:

```ts
const addListGeneric = (value: TypeBrowserObject, typeAsList: boolean, isIvyType: boolean, inCodeBlock: boolean) => {
  if (isIvyType && inCodeBlock) {
    return `List<${value.simpleName}>`;
  } 
  if (isIvyType && !inCodeBlock) {
    return `java.util.List<${value.simpleName}>`;
  }
  if (!isIvyType && inCodeBlock) {
    return `List<${value.fullQualifiedName}>`;
  }
  return `java.util.List<${value.fullQualifiedName}>`; 
}
```

or 

```ts
const addListGeneric = (value: TypeBrowserObject, typeAsList: boolean, isIvyType: boolean, inCodeBlock: boolean) => {
  let listPrefix = 'java.util.List';
  if (inCodeBlock) {
    listPrefix = 'List';
  }
  if (isIvyType) {
    return `${listPrefix}<${value.simpleName}>`;
  } 
  return `${listPrefix}<${value.fullQualifiedName}>`; 
}
```